src/Microsoft.DotNet.Build.Tasks/UpdatePackageDependencyVersion.cs

```diff
@@ -22,7 +22,7 @@ public override bool VisitPackage(JProperty package, string projectJsonPath)
             string dependencyVersion;
             if (package.Value is JObject)
             {
-                dependencyVersion = package.Value["version"].Value<string>();
+                dependencyVersion = package.Value["version"]?.Value<string>();
```

Comment:
If you mean something like `"System.Foo": {}`, then maybe we should include that in the project.json validation. It should be very rare that someone actually calls `UpdatePackageDependencyVersion` though, so I think it makes sense for this task to be lenient.